packages/aws-cdk-lib/aws-cloudfront-origins/README.md

```diff
@@ -338,6 +340,29 @@ See CloudFront docs on [Giving the origin access control permission to access th
 > Note: If your bucket previously used OAI, you will need to manually remove the policy statement
 that gives the OAI access to your bucket after setting up OAC.
 
+#### Setting up OAC with LIST permission
+
+By default, S3 origin returns 403 Forbidden HTTP response when the requested object does not exist.
+When you want to receive 404 Not Found, specify `AccessLevel.LIST` in `originAccessLevels` to add `s3:ListBucket` permission in the bucket policy.
+
+This is useful to distinguish between responses blocked by WAF (403) and responses where the file does not exist (404).
+
+``` ts
+const myBucket = new s3.Bucket(this, 'myBucket');
+const s3Origin = origins.S3BucketOrigin.withOriginAccessControl(myBucket, {
+  originAccessLevels: [cloudfront.AccessLevel.READ, cloudfront.AccessLevel.LIST],
+});
+new cloudfront.Distribution(this, 'distribution', {
+  defaultBehavior: {
+    origin: s3Origin,
+  },
+  defaultRootObject: 'index.html', // recommended to specify
+});
+```
+
+It is highly recommended to specify `defaultRootObject` distribution property.
+Without it, the root path `https://xxxx.cloudfront.net/` will return the list of the S3 object keys.
```

Comment:
Hmm this raises some concern to me, should we make it so that `defaultRootObject` MUST be specified when user adds LIST permission?

Reply:
It should not be forced to specify `defaultRootObject` because origins associated to non-default behaviors do not require it. We can add a warning annotation instead.

For example:
``` ts
new cloudfront.Distribution(this, 'Distribution', {
  defaultBehavior: {
    // the origin server will handle root path
    // setting defaultRootObject will break the web application's router
    origin: new origins.HttpOrigin('mywebserver.example.com'), 
  },
  additionalBehaviors: {
    // the bucket will not receive requests to the root path
    'assets/*': {
      origin: origins.S3BucketOrigin.withOriginAccessControl(bucket, {
        originAccessLevels: [cloudfront.AccessLevel.READ, cloudfront.AccessLevel.LIST],
      }),
    },
  },
});
```